.migrations/V20250122141100__add_cluster_routing.sql

```diff
@@ -0,0 +1,39 @@
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'tier') THEN
+CREATE TYPE tier AS ENUM ('Tier0', 'Tier1', 'Tier2', 'Tier3', 'Tier4');
+END IF;
+END$$;
+
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'cluster_status') THEN
+CREATE TYPE cluster_status AS ENUM ('active', 'maintenance', 'offline');
+END IF;
+END$$;
+
+CREATE TABLE IF NOT EXISTS cluster_state (
+    name VARCHAR PRIMARY KEY,
+    status cluster_status NOT NULL DEFAULT 'active',
+    status_reason VARCHAR,
+    status_since TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
+    capabilities JSONB NOT NULL DEFAULT '[]'::jsonb,
+    allowed_tiers JSONB NOT NULL DEFAULT '[]'::jsonb,
```

Comment:
Not 100% sure on the data model, but should this be an array of tiers?

```suggestion
    allowed_tiers tier[] NOT NULL DEFAULT '{}',
```